packages/tool-cache/src/tool-cache.ts

```diff
@@ -454,6 +455,42 @@ export function findAllVersions(toolName: string, arch?: string): string[] {
   return versions
 }
 
+// versions-manifest
+//
+// typical pattern of a setup-* action that supports JIT would be:
+// 1. resolve semver against local cache
+//
+// 2. if no match, download
+//   a. query versions manifest to match
+//   b. if no match, fall back to source if exists (tool distribution)
+//   c. with download url, download, install and preprent path
+
+export type IToolRelease = mm.IToolRelease
+export type IToolReleaseFile = mm.IToolReleaseFile
+
+export async function getManifestFromUrl(url: string): Promise<IToolRelease[]> {
```

Comment:
How confident do you feel about the behavior?

To buy leeway, I'm wondering if whether it would be good to add a function description, indicating experimental and behavior may change.

Reply:
Good suggestion.  I may go one step further and release a actions/tool-cache@preview out of this branch.  Same on the setup-node side.  Start a v2-beta